apps/web/src/app/[locale]/challenge/[slug]/_context.tsx

```diff
@@ -0,0 +1,29 @@
+'use client';
+import { createContext, useState, useContext, type PropsWithChildren } from 'react';
+
+interface TrackVisibilityContextType {
+  isTrackTitleVisible: boolean | null;
+  setIsTrackTitleVisible: React.Dispatch<React.SetStateAction<boolean | null>>;
+}
+
+export const TrackVisibilityContext = createContext<TrackVisibilityContextType | undefined>(
+  undefined,
+);
+
+export const TrackVisibiltyProvider = ({ children }: PropsWithChildren) => {
+  const [isTrackTitleVisible, setIsTrackTitleVisible] = useState<boolean | null>(null);
```

Comment:
Question: Why can't we have just boolean over there with initial value of `false`?

Reply:
lets default to false

Reply:
okay